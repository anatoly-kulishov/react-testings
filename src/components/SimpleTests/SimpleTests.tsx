import React, { useCallback, useEffect, useState } from 'react';
import axios from 'axios';

const SimpleTests = () => {
  const [data, setData] = useState('');
  const [toggle, setToggle] = useState<boolean>(false);
  const [value, setValue] = useState<string>('');
  
  const onClick = () => setToggle(!toggle);
  
  const callback = useCallback(() => {
    setTimeout(() => {
      setData('Success')
    }, 100)
  }, [])
  
  useEffect(() => {
    callback();
  }, [callback])
  
  return (
    <div>
      {data && <div className="data-wrapper" style={{color: 'red'}}>{data}</div>}
      <h1>Hello World</h1>
      <h2>{toggle && <div data-testid="toggle-elem">toggle</div>}</h2>
      <h3 data-testid="value-elem">{value}</h3>
      <button onClick={onClick} data-testid="toggle-btn">Click me</button>
      <input type="text" value={value} onChange={e => setValue(e.target.value)} placeholder="input value..."/>
    </div>
  );
};

export default SimpleTests;
