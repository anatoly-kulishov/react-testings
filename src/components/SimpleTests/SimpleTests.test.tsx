import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react';
import userEvent from "@testing-library/user-event";
import SimpleTests from "./SimpleTests";

describe('SimpleTests', () => {
  test('1', () => {
    render(<SimpleTests/>);
    
    const helloWorldElem = screen.getByText(/Hello World/i);
    const btn = screen.getByRole('button');
    const input = screen.getByPlaceholderText(/input value/i);
    
    expect(helloWorldElem).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
    expect(input).toMatchSnapshot();
  });
  
  test('2', () => {
    render(<SimpleTests/>);
    const helloWorldElem = screen.queryByText(/test/i);
    expect(helloWorldElem).toBeNull();
  });
  
  test('3', async () => {
    render(<SimpleTests/>);
    const helloWorldElem = await screen.findByText(/success/i);
    expect(helloWorldElem).toBeInTheDocument();
    expect(helloWorldElem).toHaveStyle({color: 'red'});
    expect(helloWorldElem).toHaveClass('data-wrapper');
  })
  
  test('4', () => {
    render(<SimpleTests/>);
    const toggleBtn = screen.getByTestId('toggle-btn');
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
    fireEvent.click(toggleBtn);
    // eslint-disable-next-line testing-library/prefer-presence-queries
    expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument();
    fireEvent.click(toggleBtn);
    expect(screen.queryByTestId('toggle-elem')).toBeNull();
  });
  
  test('5', async () => {
    render(<SimpleTests/>);
    const input = screen.getByPlaceholderText(/input value/i);
    expect(screen.queryByTestId('value-elem')).toContainHTML('');
    // Fire event
    fireEvent.input(input, {
      target: {value: 'firstname'}
    });
    expect(screen.queryByTestId('value-elem')).toContainHTML('firstname');
    // User event
    await userEvent.type(input, 'lastname');
    expect(screen.queryByTestId('value-elem')).toContainHTML('lastname');
  });
})

