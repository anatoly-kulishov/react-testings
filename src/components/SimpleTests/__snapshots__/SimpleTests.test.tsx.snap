// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SimpleTests 1 1`] = `
<input
  placeholder="input value..."
  type="text"
  value=""
/>
`;

exports[`TEST SimpleTests 1 1`] = `
<input
  placeholder="input value..."
  type="text"
  value=""
/>
`;
